Cargo.toml

```diff
@@ -109,6 +109,8 @@ env_logger = "0.7"
 electrsd = "0.21"
 # Move back to importing from rust-bitcoin once https://github.com/rust-bitcoin/rust-bitcoin/pull/1342 is released
 base64 = "^0.13"
+libtor = "47.8.0+0.4.7.x"
+ureq = {version = "*", features = ["json", "socks-proxy"]}
```

Comment:
Why do we need `ureq`?

Reply:
I believe BDK uses `ureq` under the hood to make HTTP requests. This line enables  `socks-proxy` feature, which is not enabled by default. This feature is required to connect to Tor. The example fails without it.

Reply:
No it doesn't.. You are using `esplora-client` internals to create a client which isn't the intended API in BDK.. You can write something like below and it would work. The `esplora-client` crate is already compiled with those features.. 

```
let esplora_config = EsploraBlockchainConfig{
        base_url: esplora_url.into(),
        proxy: Some(format!("socks5://{}", socks_addr)),
        concurrency: Some(4),
        stop_gap: 20,
        timeout: Some(60)
    };

let blockchain = EsploraBlockchain::from_config(&esplora_config).unwrap();
```

Reply:
Now I'm lost. Could you show exact code how to enable socks5 in `esplora-client`, please. This is my first BDK PR, I'm not that familiar with BDK's internals. Probably I'm missing something. 

I deleted the `ureq = ...` line from `Cargo.toml`, copy/pasted the code above, ran the example and got 
```bash
$ cargo clean
$ cargo run --features="use-esplora-blocking" --example esplora_backend_with_tor
    Finished dev [unoptimized + debuginfo] target(s) in 1.70s
     Running `target/debug/examples/esplora_backend_with_tor`
Staring Tor in /var/folders/38/pp4xfv2j7dbc6sr3gnln7jtr0000gn/T//bdk-tor-example
Dec 28 08:23:39.977 [notice] Tor 0.4.7.10 running on Darwin with Libevent 2.1.12-stable, OpenSSL 1.1.1s, Zlib 1.2.11, Liblzma N/A, Libzstd N/A and Unknown N/A as libc.
Dec 28 08:23:39.977 [notice] Tor can't help you if you use it wrong! Learn how to be safe at https://support.torproject.org/faq/staying-anonymous/
Dec 28 08:23:39.977 [notice] Configuration file ".torrc" not present, using reasonable defaults.
Dec 28 08:23:39.978 [notice] Opening Socks listener on 127.0.0.1:19050
Dec 28 08:23:39.978 [notice] Opened Socks listener connection (ready) on 127.0.0.1:19050
Tor started
Connecting to http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion/testnet/api via SOCKS5 proxy 127.0.0.1:19050
Syncing the wallet
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Esplora(Ureq(Transport(Transport { kind: ConnectionFailed, message: Some("Connect error"), url: Some(Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion")), port: None, path: "/testnet/api/scripthash/bb833461984bc2f06978e5e45c86478939c0d122625ec3b0a6a231e3c9b5378e/txs", query: None, fragment: None }), source: Some(Custom { kind: Other, error: "SOCKS feature disabled." }), response: None })))', examples/esplora_backend_with_tor.rs:69:54
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
I noticed that BDK actually uses two different `ureq` versions

```bash
   Compiling ureq v2.2.0
   Compiling ureq v1.5.5
```

Maybe that's the root cause. One of them is compiled with socks and the other is not, and for some reason the linker uses the one without socks support...

Reply:
Yeah this is a bug in `esplora-clinet` which is fixed here https://github.com/bitcoindevkit/rust-esplora-client/pull/34.. So this needs to wait for a new release of esplora-client..